core/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -188,6 +214,76 @@ public SimpleQueryBus(MessageMonitor<? super QueryMessage<?, ?>> messageMonitor,
                        .filter(Objects::nonNull);
     }
 
+    @SuppressWarnings("unchecked")
+    @Override
+    public <Q, I, U> Registration subscriptionQuery(SubscriptionQueryMessage<Q, I, U> query,
+                                                    UpdateHandler<I, U> updateHandler) {
+        MessageMonitor.MonitorCallback monitorCallback = messageMonitor.onMessageIngested(query);
```

Comment:
Don't think you should solve this asap, but a thing to think about.
Shouldn't our `MessageMonitor` set up also register updates being emitted by the `UpdateHandler`?

Reply:
This is an interesting question! The same idea crossed my mind, but wasn't sure what to do since our incremental updates are not messages per se. But, on the other hand it completely makes sense to monitor them. Would like to hear @abuijze opinion on this.